src/design/components/atoms/loaders/index.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useMemo } from 'react'
+import { useSettings } from '../../../../cloud/lib/stores/settings'
+import { selectV2Theme } from '../../../lib/styled/styleFunctions'
+import DocEditorLoader from './DocEditorLoader'
+import FolderPageLoader from './FolderPageLoader'
+import NavItemLoader, { NavItemLoaderProps } from './NavItemLoader'
+import SmartViewLoader from './SmartViewLoader'
+import TeamPickerLoader from './TeamPickerLoader'
+import TopbarLoader, { TopbarBreadcrumbLoader } from './TopbarLoader'
+
+type LoaderProps = {} & (
+  | {
+      variant: 'team-picker'
+    }
+  | {
+      variant: 'topbar'
+    }
+  | {
+      variant: 'doc-editor'
+    }
+  | ({ variant: 'nav-item' } & NavItemLoaderProps)
+  | {
+      variant: 'topbar-breadcrumb'
+    }
+  | {
+      variant: 'folder-page'
+    }
+  | { variant: 'smart-view' }
+)
+
+const commonLoaderSpeed = 6
+const commonRadiuses = 6
+const Loader = ({ variant, ...props }: LoaderProps) => {
```

Comment:
I don't think we need this component. We can use each Loader component directly where it should be placed.
Also, it will make further improvements like code splitting and tree shaking more difficult.